docs/keywords.md

```diff
@@ -0,0 +1,53 @@
+**Type of algorithm** |
+  `rule-based` | if the implementation uses a heuristic or rules based approach
+  `model-based` | if the implementation uses a machine learning model
+  `api-based` | if the implementation uses an external api or tool
+  `external-knowledge-based` | if the implementation uses a corpora, a separate knowledge base, etc.
+  `transformer-based` | if the implementation uses a transformer based model (eg. BERT, T5)
+  `parser-based` | if the implementation uses any syntactic (dependency, constituency, etc) or shallow semantic parser or other semantic parsers (eg. semantic role labelling)
+  `tokenizer` | if the implementation uses a tokenizer
+  `chunker` | if the implementation uses a chunker
+  `discourse-segmenter` | if the implementation uses a discourse segmenter
+**The naturalness of the generation** |
+  `unnatural-sounding` | if the generations might not sound natural eg. if synthesized to speech, would they not sound natural as their source sentences. Eg, adding noun definitions might make the sentence factually correct but people might not use the same language while speaking.  
+  `unnaturally-written` | generations in written format which might not look natural to a reader. homophonic translations, filler words
+**The potential accuracy & precision of the generation** | 
+  `high-precision` | The transformation generates what it intends to generate
+  `low-precision` | The transformation sometimes does not generate what it intends to generate (eg. often changes meaning)
+  `low-coverage` | The transformation might not return an output for all types of inputs but only a handful of inputs
+  `high-coverage` | The transformation can return an output for almost all types of inputs
+  `high-generations` | The transformation generates a lot of (possibly diverse) outputs for a single input (eg beam search or substitutions with large number of alternate choices)
+  `low-generations` | The transformation generates few outputs for a single input (substitution with lesser number of alternate choices)
+**The text-specific modality being addressed (if applicable)** |
+  `visual` | The outputs are visually similar to the inputs (eg. Leet letters)
+  `aural` | The outputs sound similar to the inputs (eg. homophones)
+  `written` | The transformation makes font changes, or changes which would look different in different written formats
+**The skills required to differentiate between the input and the output of the transformation (if applicable)** |
+  `causal-reasoning` | ability to reason about cause and effect
+  `visual-reasoning` | ability to reason by the appearance of the text
+  `social-reasoning` | ability to understand and reason about human social interactions
+  `figurative-language` | ability to understand figures of speech
+  `world-knowledge` | ability to understand world knowledge
```

Comment:
Would it make sense to somehow merge modality into this category? I feel we are having too many categories, and e.g. `visual reasoning` would closely overlap with `written`, `visual`, or `unnaturally written` in previous sections.

Reply:
You are right. I've removed visual-reasoning now for now.